packages/@coorpacademy-components/src/molecule/drag-and-drop/upload-report.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  NovaCompositionCoorpacademyValidate as Validated,
+  NovaCompositionCoorpacademyBrokenHeart as BrokenHeart
+} from '@coorpacademy/nova-icons';
+import get from 'lodash/fp/get';
+
+import {pipe, split, last} from 'lodash/fp';
+import Provider from '../../atom/provider';
+
+import style from './upload-report.css';
+import {Preview} from './preview';
+
+export const UploadReport = (
+  {error, previewContent, previewLabel, uploadErrorMessage, uploadSuccessMessage},
+  context
+) => {
+  const translate = get('translate', context);
+
+  if (error) {
+    return (
+      <div className={style.reportingContainer}>
+        <div className={style.repport}>
+          <span> {translate('sad_face')}</span>
```

Comment:
```suggestion
          <span> {translate('sad_face:emoji')}</span>
```

Reply:
en mode error and not on a dans les 2 cas ce template:
```
<div className={style.reportingContainer}>
        <div className={style.repport}>
            xxxxx
        </div>
       <div className={style.previewContainer}>
              yyyy
        </div>
</div>
```
A refactor?

Reply:
Je pensais aussi factoriser ce composant mais après essai, le coeur du composant ressemblerait à ça : 

`
    <div id="upload-report" className={style.reportingContainer}>
      <div className={style.repport}>
        <span> {error ? translate('sad_face') : translate('tada')}</span>
        <div>
          <p className={style.label}>{error ? uploadErrorMessage : uploadSuccessMessage}</p>

          {error ? (
            <div>
              <p className={style.label}>{uploadErrorMessage}</p>
              <p className={style.uploadErrorMessage}>{error}</p>
            </div>
          ) : (
            <p className={style.fileName}>
              {pipe(
                split('/'),
                last
              )(previewContent.src)}
            </p>
          )}
        </div>
        <Validated className={style.reportIcon} />
      </div>
      <div className={style.previewContainer}>
        {error ? (
          <BrokenHeart className={style.brokenHeartIcon} />
        ) : (
          <Preview {...previewContent} />
        )}
      </div>
    </div>
`

De mon point de vue, je pense que la gain de LOC ne compense pas la perte en lisibilité